### PR TITLE
Dropdown value pairs

### LINKED ISSUE #362: Separate value from display name in dropdown component
**Is your feature request related to a problem? Please describe.**

Alternative answers in AHV includes both value and display name. I.e. if we need to select a country, we could have a model like the following:

```
const choices = [
  { "value": 1, "name": "Norge" },
  { "value": 2, "name" "Sverige" },
]
```

In this case, we'd like to display `name` to the user, whilst we use `value` internally. In Jøkul, we have to index values by name and translate back and forth to handle changes:

```
const valuesIndexedByName = choices.reduce((acc, { value, name }) => ({
        ...acc,
        [name]: value,
    }), {});

<Dropdown
        label=""
        items={Object.keys(valuesIndexedByName)}
        onChange={name => answerQuestion(valuesIndexedByName[name])}
    />
```

**Describe the solution you'd like**

Ideally, the dropdown component should support value and display value fields so that we could render the component as shown below.

```
<Dropdown
        items={choices}
        onChange={value => answerQuestion(value)}
    />
```

Additionally, we don't want to submit a label field into the component, this should be optional. In our case the label is rendered elsewhere (see #330).

**Describe alternatives you've considered**

We currently translate between text and value outside the component, it works, but it's not a good solution.

### PR DESCRIPTION
**Pull request by piofinn**
## 📥 Proposed changes

Let `Dropdown` and `Select` accept value/label pairs in addition to `string`s as items. This matches native behaviour better, and closes #362 .

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

## 💬 Further comments

The form of the value/label pairs is defined as a type (`SelectValuePair`) in `SelectValuePair.ts` and can be imported from there. Both `Dropdown` and `Select` accepts an array of _either_ `string`s or `SelectValuePair`s as the `items` prop.

```ts
type SelectValuePair = {
    value: string;
    label: string;
};
```
```ts
interface Props {
    ...
    items: Array<string | SelectValuePair>;
}
```

### CODE REVIEW

File: packages/dropdown-react/src/Dropdown.tsx [hunk 1]
```diff
@@ -5,9 +5,14 @@ import nanoid from "nanoid";
 import { SupportLabel } from "@fremtind/jkl-typography-react";
 import { useListNavigation } from "./useListNavigation";
 
+export type SelectValue = {
```
Comment (piofinn) [inline packages/dropdown-react/src/Dropdown.tsx]: Should this be exported from `index.ts` as well?

**Comment by piofinn** [comment]
@stianlik, does this solve your issue? I can't request your review since you're not in the organization, but please take a look! 😊

**Comment by stianlik** [comment]
@piofinn Yes, this is exactly what we need, thanks :)

File: packages/dropdown-react/src/Dropdown.tsx (hunk 1 above)
Comment (lfbergee) [reply packages/dropdown-react/src/Dropdown.tsx]: Not a huge fan of the name. SelectValuePair or SelectObject maybe

File: packages/dropdown-react/src/Dropdown.tsx [hunk 2]
```diff
@@ -78,7 +87,7 @@ export function Dropdown({
         e.target.button.focus();
         e.target.value = e.detail;
         setSelectedValue(e.detail.textContent);
-        onChange && onChange(e.detail.textContent);
+        onChange && onChange(selectedValue || "");
```
Comment (lfbergee) [inline packages/dropdown-react/src/Dropdown.tsx]: Can we end up with a race condition here? Perhaps safer to use e.detail.textContent.

```
const nextSelectedValue = e.detail.textContent;
setSelectedValue(nextSelectedValue)
onChange(nextSelectedValue || "")
```

File: packages/dropdown-react/src/Dropdown.tsx [hunk 3]
```diff
@@ -40,6 +45,10 @@ function focusSelected(listEl: HTMLElement, listId: string, selected: string | u
     focusedItem && focusedItem.focus();
 }
 
+export function makeSelectValueFrom(item: string | SelectValue) {
```
Comment (lfbergee) [inline packages/dropdown-react/src/Dropdown.tsx]: In my mind `make` is a keyword for a function that returns a function. I'd prefer `getSelectValue()` here

Comment (lfbergee) [reply packages/dropdown-react/src/Dropdown.tsx]: And since both uses this, i'd prefer it to be split out to it's own file or utility file

Comment (lfbergee) [reply packages/dropdown-react/src/Dropdown.tsx]: And write a small litte test for it :dancer:

File: packages/dropdown-react/src/Dropdown.tsx (hunk 2 above)
Comment (piofinn) [reply packages/dropdown-react/src/Dropdown.tsx]: I was going to say that `e.detail.textValue` is the wrong value here, which it is, but then I saw that it is used as it is now as well 😑 We want to return the `value` from the value pair, but the text content is the `label`. I'll take a look at getting this right.

**Pull request merged**